otwcode/otwarchive
# AO3-5519 AO3-5883 AO3-5886 Restrict admin access to admin posts, settings, and comments

# Pull Request Checklist

  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-5519
+
https://otwarchive.atlassian.net/browse/AO3-5883
+
https://otwarchive.atlassian.net/browse/AO3-5886


## Purpose
Limits admins ability to do certain actions, based on the roles they've been assigned in the DB.

BUILDS ON PR #3821.

[Diff for new commits](https://github.com/otwcode/otwarchive/pull/3836/files/86a3dfbf4185073e6d119b9b5390596225ddb413..675a33464d9160e5405f39409e126df47ba06c38) for ease of reviewing

## Testing Instructions

5519:
SIgn in with an admin user with superadmin, communications, or translation role, and ensure you can access/ use the following:
“Post AO3 News” option under “Admin Posts” in the admin navigation
“Edit Post” or “Delete Post” options on the news posts themselves
admin post index when following the “AO3 News” link under “About” in the main navigation, or otherwise attempting to access the URL for the admin posts index (https://test.archiveofourown.org/admin_posts)

If they do not have this role,. they should be redirected and given an error message if they enter the URL to post (https://test.archiveofourown.org/admin_posts/new) or edit (https://test.archiveofourown.org/admin_posts/360/edit) an admin post in their browser

5883:
Sign in with an admin user with no rules and ensure the settings button is missing from the header in the admin section.

Sign in with a super admin and ensure that you can update all of the settings.

Sign in with an admin with only the tag_wrangling role, and ensure that you can only update the "Turn off tag wrangling for non-admins" setting.

Sign in with an admin with only the support role, and ensure that you can only update the "Turn off support form" setting.

Sign in with an admin with only the policy_and_abuse role, and ensure that you can only update the "Automatically hide spam works" setting.

5886:
Sign is as admin without appropriate roles and check that the "spam" and "delete" buttons do not exist on comments.

Sign is as an admin with correct roles. Verify the "spam" and "delete" buttons exists, and these buttons function as expected.

## References

Are there other relevant issues/pull requests/mailing list discussions?

## Credit

*What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?*

(If you have a Jira account, please include the same name in the "Full name"
field on your Jira profile, so we can assign you the issues you're working on.)

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*

(If you do not fill in this section, we will use your GitHub account name and
they/them pronouns to avoid making assumptions about your gender.)
